src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/RegenerateConditionalOrderPronouncedCoverLetter.java

```diff
@@ -25,54 +25,60 @@ public CaseDetails<CaseData, State> apply(CaseDetails<CaseData, State> caseDetai
 
         final Long caseId = caseDetails.getId();
         final CaseData caseData = caseDetails.getData();
-
         if (caseData.getApplicant1().isApplicantOffline() && caseData.getApplicant1().isConfidentialContactDetails()) {
```

Comment:
Could you explain why the isConfidentialContactDetails() check happens here? Don't we need to regenerate for all broken cases?

Reply:
that's just me refactoring the existing class because sonar complained - I did want to leave it alone but there was too much duplication, so I've carefully moved the code to statics in this one and then call from the other.